src/views/networkpolicies/new/NetworkPolicyPortsDropdown.tsx

```diff
@@ -3,44 +3,44 @@ import React, { FC, Ref, useState } from 'react';
 import { Dropdown, DropdownItem, MenuToggle, MenuToggleElement } from '@patternfly/react-core';
 import { NetworkPolicyPort } from '@utils/models';
 
-type PortsDropdownProps = {
+type NetworkPolicyPortsDropdownProps = {
   index: number;
   onSingleChange: (port: NetworkPolicyPort, index: number) => void;
   port: NetworkPolicyPort;
 };
 
-const PortsDropdown: FC<PortsDropdownProps> = ({ index, onSingleChange, port }) => {
-  const key = `${port.key}-${index}`;
-
+const NetworkPolicyPortsDropdown: FC<NetworkPolicyPortsDropdownProps> = ({
+  index,
+  onSingleChange,
+  port,
+}) => {
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
 
   return (
     <Dropdown
       data-test="port-protocol"
       isOpen={isDropdownOpen}
```

Comment:
Please add `onOpenChange(setIsDropdownOpen}` property,
This will allow closing the dropdown when it's open and you click outside the dropdown

Reply:
ok

Reply:
next pr